### PR TITLE
Python 3 and PyPy support.

### PR DESCRIPTION
**Pull request by jamadden**
Stop publishing INSTALL.txt on PyPI because it's very misleading at
this point.

Remove zope.app.testing and zope.app.zcmlfiles dependencies.

Test coverage could be better, I'll see what I can do easily.

### CODE REVIEW

**Comment by jamadden** [comment]
Can we drop the stuff in zopeversion.py that deals with trying to find the version from a SVN checkout? That shouldn't be possible anymore.

**Comment by jamadden** [comment]
The SVN stuff was quite probably broken, depending on exactly what path it was given, due to the change in the layout of SVN working directories (to have one central .svn, like .git). That code path could not be taken in the singleton instance as there is no path given, and I found no zopefoundation code that used it with a path. So I think it was safe to remove it; it looks like it had been expected to return `Meaningless` since 3.5.9 from late 2010 anyway.

That and some extra tests get coverage up to 100%. I think it looks pretty good now.

**Comment by jamadden** [comment]
Heh. So I was able to drop the zope.app.rotterdam dependency, which was cool because it meant I could use z3c.pt (rotterdam depends on implicit variable names like 'macroname' that z3c.pt doesn't define), which was cool because it uses Chameleon, which is cool because coverage.py supports plugins for things like templating languages, which is cool because there's such a plugin for Chamelean, which meant we could work up a standard way to get coverage reports for templates...

...which turned out to be not cool, because I guess I just dreamed the part where there was a Chameleon plugin. The closest I could find was https://github.com/malthe/chameleon/issues/193

Ah well, at least we dropped some deps.

File: src/zope/app/applicationcontrol/browser/runtimeinfo.py [hunk 1]
```diff
@@ -12,15 +12,17 @@
...
 __docformat__ = 'restructuredtext'
 
 from zope.app.applicationcontrol.interfaces import IRuntimeInfo
 
 from zope.app.applicationcontrol.i18n import ZopeMessageFactory as _
 
+try:
+    long
+except NameError:
+    long = int
```
Comment (mgedmin) [inline src/zope/app/applicationcontrol/browser/runtimeinfo.py]: You could just use `int` on Python 2 as well.  We're not supporting Python 2.1 any more ;)

File: src/zope/app/applicationcontrol/browser/zodbcontrol.py [hunk 2]
```diff
@@ -13,39 +13,46 @@
...
-            d = dict(
-                dbName = db.getName(),
-                utilName = str(name),
-                size = self._getSize(db),
-                )
+        for name, db in component.getUtilitiesFor(IDatabase):
+            d = {
+                'dbName': db.getName(),
+                'utilName': str(name),
+                'size': self._getSize(db)
```
Comment (mgedmin) [inline src/zope/app/applicationcontrol/browser/zodbcontrol.py]: I like trailing commas in dict literals when the closing `}` is on the next line.

File: src/zope/app/applicationcontrol/zopeversion.py [hunk 3]
```diff
@@ -52,60 +44,13 @@ def getZopeVersion(self):
...
-                try:
-                    self.result = f.readline().strip() or self.result
-                finally:
-                    f.close()
-        return self.result
+        # try to get official Zope release information
+        versionfile = os.path.join(self.path, "version.txt")
+        if os.path.isfile(versionfile):
+            with open(versionfile) as f:
+                self.result = f.readline().strip() or self.result
```
Comment (mgedmin) [inline src/zope/app/applicationcontrol/zopeversion.py:51]: I think this bit can be removed -- a single `version.txt` to define the version of Zope 3 is a legacy from when we had a monolithic Subversion repository, before the eggsplosion.  It no longer makes sense.

But this leaves us with nothing here (or, rather, a constant 'Development/Unknown' "version").

I think what would make sense is to enumerate all the packages known to setuptools/pkg_resources and show their names and versions.  But that's a separate task, not a part of the Python 3 port PR.

**Review by mgedmin — APPROVED** [review]
LGTM!

**Comment by mgedmin** [comment]
You now have PyPI rights to zope.app.applicationcontrol

File: src/zope/app/applicationcontrol/browser/runtimeinfo.py (hunk 1 above)
Comment (jamadden) [reply src/zope/app/applicationcontrol/browser/runtimeinfo.py]: Oh, cool, I didn't realize Python 2 would handle the overflow automatically. TIL!

```pycon
>>> sys.maxint + 1
9223372036854775808L
>>> i = 9223372036854775808L
>>> int(i)
9223372036854775808L
```

File: src/zope/app/applicationcontrol/browser/zodbcontrol.py (hunk 2 above)
Comment (jamadden) [reply src/zope/app/applicationcontrol/browser/zodbcontrol.py]: D'oh! Me too. Thanks.

File: src/zope/app/applicationcontrol/zopeversion.py (hunk 3 above)
Comment (jamadden) [reply src/zope/app/applicationcontrol/zopeversion.py:51]: I agree, it's pretty meaningless in most cases. There's a big comment to that effect up in `__init__`, which winds up setting the result to `Meaningless` so this code never even runs in production.

But hey, there were test cases for it, and I suppose some code somewhere could explicitly be setting a path.

I'll open an issue for doing something better in the no-path-given case (but probably won't get to it for a long time personally.)

**Comment by jamadden** [comment]
Thanks for the review! (And the access---it's almost like I'm predictable or something 😄 )

**Pull request merged**